### ISSUE TITLE
Not an issue  . . . . yet. just an inquiry.

### ISSUE THREAD

**Issue by ineuw** [autokey/autokey#425]
## Classification:
Information request

## The question
I store a set of Autokey 95.10 installation files in both versions, downloaded from Github. I use the GTK version and on installation, I got a message from DebInstaller, that there is a newer version available in the Software Manager. So, instead of my copy, I installed from the software manager. to see if there is a difference. Is there any?

**Comment by josephj11** [comment]
There is really only one version of AutoKey at each release level. It just has two front ends, one for GTK and another for Qt. Both front ends can be installed on the same system at the same time. You just have to choose which one to launch when you start AutoKey.

I don't know what your software manager has in its repository. What distribution/release are you using?

Typically distributions just repackage applications in their packaging systems so that they install correctly on the target distribution. For a few major packages (certainly not AutoKey) they may also make small modifications to the packages.

Given that it took a lot of time and effort to get the major distributions to even include packages of the current version of AutoKey, its pretty unlikely that they would spend additional effort customizing it. It's also not particularly obvious what they might even want to customize. If they wanted something reasonable and just asked us, we'd probably implement it - although I can't speak for our developers.

Bottom line: Start AutoKey and open the Main menu, select Help->About, and see if the version number displayed is the one you expect. If it is, then it is very likely that all is well. If it's not, then the distribution/repository you are using is doing something strange.

In that case, we have no influence on what they do or how they do it, so you can either contact them or just stop using their repository.

BTW: Whenever you have a question (as opposed to an issue) it's better to post it on our list or on Gitter. The list is more user oriented whereas more technical questions will get a better answer on Gitter.

[list](https://groups.google.com/forum/#!forum/autokey-users)

[Gitter](https://gitter.im/autokey/autokey)

**Comment by ineuw** [comment]
Thanks for letting me know about gitter. Does this mean that google site is
moving over? As for the version numbers, they are the same. except Ubuntu
may have changed some things because it crashed hence the reason I asked.

ineuw

**Comment by josephj11** [comment]
If you can document a specific case which works on the version of AutoKey from our source or deb files here, but fails on the same (hopefully, the latest) version of AutoKey installed from some other reputable source, that's a real issue. I'm not sure how we would address it, but we would want to know about it.

As I mentioned, our list is better for general questions and Gitter is better for technical questions. The main distinction is that our developers do not post on the list, but they do on Gitter. So when the few of us "experts" can't answer something on the list, then the developers have to be asked so that someone who actually knows how things work and what is possible can answer.

In order to keep the developers developing instead of answering simple (and often duplicate) questions, it's good for us to have both venues. There is no enforcement of this distinction, but it tends to work in practice.

What I don't want to see (and I'd guess the developers would agree) is our list of actual issues (here) getting cluttered with non-issues. I often search the open (and sometimes closed) issues when I see something that looks familiar and I want to see if relevant issues already exist and this is where the developers track the progress of the project and decide what they're going to tackle next. Keeping this issue list focused on real issues makes such processes much more efficient.

From a purely practical standpoint, many more people read and reply to our list and to Gitter posts than they do for issues, so you're much more likely to get more answers more quickly when you ask in the right place. And, for the same reason, more people will benefit from being able to see these discussions which may answer questions they may have or give them new ideas about how to use AutoKey.

**Comment by ineuw** [comment]
I find three forums to be disconcerting. I prefer Gitter to Google Groups
and it was just a case of old habits, posting in the wrong place. It would
be advantageous to all GG participants to move to Gitter as well because of
everything you pointed out. Every topic posted to GG is related to one
software. Programming questions, feature discussions and application
questions should all be on Gitter, including non "issue" related matters
from Github.

I will gladly test again and record any differences between the two
sources. Although they have the same version number, I didn't keep a
written record of the differences.

**Comment by josephj11** [comment]
Let's agree to disagree. :)

**Comment by ineuw** [comment]
Hi,

Can you elaborate about the disagreement? Why are three sites better than
two?

Returning to the subject of the two versions of the software, I am keeping
notes. I am guessing that qt is the older version? And may be the more
popular version? There are differences. Some cosmetic, but so far I've only
compared a few of my keyboard hotkeys, and I take notes of my observations.
For multiple reasons, I am enjoying being occupied with Linux, Python and
Autokey, instead of Wikipedia/Wikisource where I haven't done any work for
weeks.

Be well,

**Comment by josephj11** [comment]
As I detailed above, I see three overlapping domains.

Our list handles newbie and general questions as well as some scripting issues. This is generally not followed by our developers and doesn't distract them from developing - especially because Thomas is particularly generous with his time.

Gitter is where the developers communicate with each other and are available for advanced questions and for discussing things like feature requests or possible bugs before actual issues are filed.

GitHub is (or should be) strictly for reporting and tracking bugs and feature requests.

This is a hierarchy where each successive level has less volume. It filters out things at the lowest possible level where the most people are available to handle them and conserves our most scarce resource - developer expertise and time.
----
Programming will do that to you. Once you get a taste of being able to get things done your way, a million possibilities emerge and it's hard to resist all of them. Sooner or later, the pendulum swings back the other way and you want to get something done besides tweaking your computer. Then, the cycle repeats. :)

**Comment by josephj11** [comment]
@ineuw  I don't know how the bigger font got in my last reply. It wasn't intentional! Hope you didn't think I was shouting at you. Sorry!

**Comment by ineuw** [comment]
Haha, I didn't notice it until you pointed it out. :-)
